database/apid/src/main/java/com/example/javaopencommerce/item/ItemEntity.java

```diff
@@ -29,4 +48,18 @@
     @Builder.Default
     private List<ItemDetailsEntity> details = new ArrayList<>();
 
+    Item toItemModel() {
+        List<ItemDetails> detailsModels = details.stream()
```

Comment:
It's not null-safe as well.
```suggestion
        List<ItemDetails> detailsModels = Optional.ofNullable(details).orElse(Collections.emptyList()).stream()
```

Reply:
Fixed!